src/openrct2/world/Banner.cpp

```diff
@@ -93,7 +93,7 @@ static uint8_t banner_get_ride_index_at(int32_t x, int32_t y, int32_t z)
         if (ride == nullptr || ride_type_has_flag(ride->type, RIDE_TYPE_FLAG_IS_SHOP))
             continue;
 
-        if ((tileElement->GetClearanceZ()) + 32 <= z)
+        if ((tileElement->GetClearanceZ()) + COORDS_XY_STEP <= bannerCoords.z)
```

Comment:
Z is not X or Y, so don't use COORDS_XY_STEP.

```suggestion
        if ((tileElement->GetClearanceZ()) + (4 * COORDS_Z_STEP) <= bannerCoords.z)
```

Reply:
Argh, my bad. Fixed it!